truffleruby/src/main/ruby/core/library.rb

```diff
@@ -48,20 +48,22 @@ def attach_function(name, _, _, _=nil, _=nil)
 
     suffixes = ['rubysl/rubysl-socket/lib/rubysl/socket.rb']
 
-    suffixes.each do |suffix|
-      if caller.end_with?(suffix) and Truffle::POSIX.respond_to?(mname)
-        define_method mname, &Truffle::POSIX.method(mname)
-        module_function mname
-        return
-      end
+    suffix = suffixes.find do |s|
+      caller.end_with?(s) and Truffle::POSIX.respond_to?(mname)
```

Comment:
It's our own code and it make more sense to check before `Truffle::POSIX.method(mname)`.
This code could be much simplified since there is only one `suffixes`.